app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1891,8 +1891,10 @@ public void onFullscreenStateChanged(final boolean fullscreen) {
 
         if (fullscreen) {
             hideSystemUiIfNeeded();
+            viewPager.setVisibility(View.GONE);
```

Comment:
Uhm, what is this change?

Reply:
this avoids that the comment / next video page is reachable even though the video is playing in fullscreen

Reply:
I think that was intended, wasn't it @avently?

Reply:
but it doesn't make sense for an invisible/overlayed layout to be focusable. I don't think this was intended.

Reply:
@Stypox, thank you for letting me know.
@Redirion you just removed an ability to view comments, description, etc by long pressing on arrow down button. Now I'm unable to view anything because viewpager is hidden. 
The behavior that was before is intended. Please, remove this line or find a workaround for situation you're trying to solve that doesn't ruin the intended ability.

Reply:
I am sorry then. Was too quick

Reply:
No problem. You probably didn't use/know that feature so it's expected

Reply:
@avently 
Possible solution is to only enable this feature on non-Android TV devices (as in the accepted answer here: https://stackoverflow.com/questions/27138838/how-can-i-check-if-an-app-is-running-on-an-android-tv).
What do you think? This is really *not* needed on phones or tablets.

Reply:
@mbarashkov why did you introduce this change in the first place? What issue does it solve? I mean, for me the fact that we can scroll down to comments is never in the way

Reply:
@Stypox 
On Android (non-TV) devices with a physical keyboard (BT keyboard or relatively rare devices like Cosmo Communicator) it would be very useful to press space to pause/unpause video.
Doing it while the comments view is visible is impossible, because it gets first focus.
Also, it's really very confusing to press down while in full screen and see comments - it really looks like a bug. I agree it can be useful on TV devices, of course, but not on tablets/phones.

Reply:
This is needed on phones or tablets, it was introduced since it could be useful to view description or comments while watching a video, not only on android tvs

Reply:
Well.. I was honestly confused by this. I'm in full screen, I press down and I'm in comments. 
And I **can't go back to the video at all.**
This is why I thought it's an incorrect behavior.
What do you think? @Stypox

Reply:
In this case, you need to use this method to check whether NewPipe is running on a TV:
https://github.com/TeamNewPipe/NewPipe/blob/3c67df263c494ba437d4788f7208e41aaaabed43/app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java#L24

Reply:
@mbarashkov
Take a look at how we use the arrow button:
[ScreenRecord-2021-01-10-15-07-37.mp4.zip](https://github.com/TeamNewPipe/NewPipe/files/5792554/ScreenRecord-2021-01-10-15-07-37.mp4.zip)

>it would be very useful to press space to pause/unpause video.
Doing it while the comments view is visible is impossible, because it gets first focus and onKeyDown event is lost

Event can not be lost. Find it and intercept. Move focus to play/pause button after that if needed. There is already working code for TVs which react on DPAD buttons without problems. Take a look on it too

Reply:
Well DPAD is handled in VideoPlayerImpl -> onKeyDown
Issue is: onKeyDown is never even being called (on any hardware keyboard button press) when the comments ViewPager is shown in fullscreen mode.